test/framework/vspherecsi.go

```diff
@@ -14,24 +21,33 @@ const (
 	kubeSystemNameSpace        = "kube-system"
 )
 
+var maxRetries = 5
+
 // ValidateVSphereCSI checks whether vsphere csi exists as expected or not.
 func (e *ClusterE2ETest) ValidateVSphereCSI(installed bool) {
 	ctx := context.Background()
-	_, err := e.KubectlClient.GetDeployment(ctx, csiDeployment, kubeSystemNameSpace, e.cluster().KubeconfigFile)
+	if !installed {
+		maxRetries = 1
```

Comment:
this is updating a package variable, we shouldn't do that
Updating a global state from running code is dangerous
Just define `defaultGetMaxRetries` as a constant and then define a variable on this method that you can update, initializing it to the const value